File: Java/src/test/java/com/hamaraBasket/NonExpiryItemQualityCheckerTests.java
package com.hamaraBasket;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class NonExpiryItemQualityCheckerTests {
    @Test
    public void indianWineQualityShouldIncreaseByOne() {
        HamaraBasketItem item = new HamaraBasketItem(10, 10, ItemType.NonExpiry);
        QualityChecker checker = new NonExpiryItemQualityChecker();

        checker.updateQuality(item);

        assertEquals(11, item.getQuality());
    }

    @Test
    public void indianWineQualityShouldNeverBeMoreThan50() {
        HamaraBasketItem item = new HamaraBasketItem(10, 50, ItemType.NonExpiry);
        QualityChecker checker = new NonExpiryItemQualityChecker();

        checker.updateQuality(item);

        assertEquals(50, item.getQuality());
    }
}

File: CSharpCore/HamaraBasketTests/NonExpiryItemQualityCheckerTests.cs
using HamaraBasketQualityAssurance;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace HamaraBasketTests
{
    [TestClass]
    public class NonExpiryItemQualityCheckerTests
    {
        [TestMethod]
        public void IndianWineQualityShouldIncreaseByOne()
        {
            var item = new HamaraBasketItem(10, 10, ItemType.NonExpiry);
            var checker = new NonExpiryItemQualityChecker();
            
            checker.UpdateQuality(item);

            Assert.AreEqual(11, item.Quality);
        }

        [TestMethod]
        public void IndianWineQualityShouldNeverBeMoreThan50()
        {
            var item = new HamaraBasketItem(10, 50, ItemType.NonExpiry);
            var checker = new NonExpiryItemQualityChecker();
            
            checker.UpdateQuality(item);

            Assert.AreEqual(50, item.Quality);
        }
    }
}
